R/runapp.R

```diff
@@ -467,7 +467,9 @@ runExample <- function(example=NA,
                        launch.browser = getOption('shiny.launch.browser', interactive()),
                        host=getOption('shiny.host', '127.0.0.1'),
                        display.mode=c("auto", "normal", "showcase")) {
-  examplesDir <- system_file('examples', package='shiny')
+  legacy <- getOption('shiny.legacy.examples', FALSE)
+  examplesDir <- if (isTRUE(legacy)) 'examples' else 'shiny'
+  examplesDir <- system_file(examplesDir, package='shiny')
```

Comment:
@karangattu started working on extending `runExample()` to other packages in #3917.

That said, while we're here, I think we should do a very light-weight version of that work and add `package = "shiny"` to `runExample()`'s parameters.

I'll take that in a follow-up PR.